### PR TITLE
SNOW-1754879: Improve documentation for Series.map to reflect the unsupported features

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1754879

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Improve documentation for Series.map to reflect the unsupported features.

### CODE REVIEW

**Review by sfc-gh-nkrishna — COMMENTED** [review]
Does the API documentation for `map` need updating as well?

**Comment by sfc-gh-helmeleegy** [comment]
> Does the API documentation for `map` need updating as well?

Good point. I added a small change there too.

File: src/snowflake/snowpark/modin/plugin/docstrings/series.py [hunk 1]
```diff
@@ -1833,7 +1835,7 @@ def map():
 
         ``map`` accepts a ``dict`` or a ``Series``. Values that are not found
         in the ``dict`` are converted to ``NaN``, unless the dict has a default
-        value (e.g. ``defaultdict``):
+        value (e.g. ``defaultdict``) (Currently not supported by Snowpark pandas):
```
Comment (sfc-gh-jjiao) [inline src/snowflake/snowpark/modin/plugin/docstrings/series.py:1838]: the reason we want to keep those unsupported things in the doc is we don't want to forget about thos?

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 - Improved generated SQL query for `head` and `iloc` when the row key is a slice.
 - Improved error message when passing an unknown timezone to `tz_convert` and `tz_localize` in `Series`, `DataFrame`, `Series.dt`, and `DatetimeIndex`.
+- Improved documentation for `Series.map` to reflect the unsupported features.
```
Comment (sfc-gh-jjiao) [inline CHANGELOG.md:24]: also the document for the constructor?

**Review by sfc-gh-jjiao — APPROVED** [review]

**Review by sfc-gh-nkrishna — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/docstrings/series.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/docstrings/series.py:1838]: That's correct.

**Pull request merged**